src/configuration/ConnectionSettings/DruidAuthSettings.tsx

```diff
@@ -22,8 +29,11 @@ export const DruidAuthSettings: FC<ConnectionSettingsProps> = (props: Connection
         `}
       >
         <Field horizontal label="With basic authentication" description="Enable HTTP Basic authentication">
-          <Switch value={settings.basicAuth} onChange={onSettingChange} />
+          <Switch value={settings.basicAuth} data-testid="basicAuth" name="basicAuth" onChange={onSettingChange} />
```

Comment:
we gonna need to add such `data-testid` on elements ? is there any alternative ?

Reply:
This is one way I found to test, I think as we gain more experience with the react-testing lib we will get better and maybe find a better way too. :)
I have seen other libraries do this too, so probably not the worst thing? (I guess we will know)

Reply:
See https://github.com/grafadruid/druid-grafana/pull/40#discussion_r587296687